src/components/screens/ScreenChapterStudent.tsx

```diff
@@ -439,185 +451,175 @@ export const ScreenChapterStudent: React.FC = () => {
-                        //       not update otherwise, we lose the last written progress
-                        //       in our local instance.
-                        /*if (hasCreated)*/
-                        updateCourseProgress();
-                    }
-                );
+                if (status.durationMillis !== undefined && status.durationMillis > 0) {
+                    const progress = status.positionMillis / status.durationMillis;
+
+                    ProgressService.getInstance().updateContentProgress(
+                        course.id,
+                        currentVideo,
+                        progress,
+                        (newContentProgress, hasCreated) => {
+                            // TODO: This should only update on hasCreated, but if we do
+                            //       not update otherwise, we lose the last written progress
+                            //       in our local instance.
+                            /*if (hasCreated)*/
+                            updateCourseProgress();
+                        }
+                    );
+                }
 
                 setVideoCompletedIfNecessary(currentVideo, status);
             }
         }
     }
 
-    /** Sets the video progress as completed if it matches the criteria. */
-    async function setVideoCompletedIfNecessary(contentRef: IContent, status: AVPlaybackStatus) {
-        const contentProgress: IContentProgressTracker = courseProgress.contentProgressTrackers[contentRef.id];
-        if (contentProgress != undefined && contentProgress.state != ContentProgressTrackerState.COMPLETED) {
-            const position = status["positionMillis"];
-            const duration = status["durationMillis"];
-            if (duration != undefined && position + 15000 > duration) {
-                completeVideo(contentRef);
-            }
-        }
-    }
-
     /** Restores the watch progress of the current video. */
     function restoreWatchProgress() {
         if (
-            courseProgress == undefined ||
-            courseProgress.contentProgressTrackers == undefined ||
-            currentVideo == undefined ||
-            currentVideo.id == undefined
+            courseProgress === undefined ||
+            courseProgress.contentProgressTrackers === undefined ||
+            currentVideo === undefined ||
+            currentVideo.id === undefined
         ) {
             return;
         }
-        let progress = courseProgress.contentProgressTrackers[currentVideo.id]?.progress;
-        if (progress == undefined) progress = 0;
-        videoPlayer.current?.setPositionAsync(progress * 1000);
+
+        const progress = courseProgress.contentProgressTrackers[currentVideo.id]?.progress ?? 0;
+
+        const vidPlayerInst = videoPlayer.current;
+        vidPlayerInst?.getStatusAsync().then((status) => {
+            if (status.isLoaded) {
+                const millisToSet = progress * (status.durationMillis ?? 0);
```

Comment:
Habt ihr das jetzt doch auf % statt nem absoluten Wert in Sekunden umgebaut?

Reply:
Mussten wir wohl oder übel, damit man eine richtige Progressbar innerhalb der Videokacheln anzeigen lassen kann, da wir im Frontend nicht ohne Umwege an die Länge eines Videos kommen, afaik. Funktioniert trotzalledem wie zuvor.

Reply:
Ja, fand ich irgendwie sowieso ein bisschen praktischer. Glaube der Verlust an Auflösung bei der Umrechnung bei langen Videos ist vernachlässigbar.